### ISSUE TITLE
Vista ctags can't handle nested classes with stuff in them

### ISSUE THREAD

**Issue by idanarye** [liuchengxu/vista.vim#70]
**Describe the bug**
When I have a class inside a class, and have things inside that inner class, Vista doesn't work.

**Environment:**
- OS: ArchLinux
- Vim/Neovim version: Neovim 0.3.4 release
- This plugin version: 1ea2ca78b782eefb897fbc1a9fa7e5d433436341

**Vista info**

```
    Current FileType: text
Avaliable Executives: ['ctags', 'lcn']
    Global Variables:
    let g:vista#executives = ['coc', 'ctags', 'lcn', 'vim_lsp']
    let g:vista_ctags_renderer = 'default'
    let g:vista_fzf_preview = ['right:50%']
    let g:vista_stay_on_open = 0
```

**Steps to reproduce given the above info**
1. Create this file:
    ```python
    class Foo:
        class Bar:
            def baz(self):
                pass
    ```
2. Run `:Vista ctags`

**Expected behavior**
Vista's sidebar should open

**Actual behavior**
A blank sidebar and this error message:

```
Error detected while processing function <SNR>293_on_exit[6]..<SNR>293_ApplyExtracted[14]..vista#viewer#Display[1]..<SNR>299_Render[2]..vista#renderer#default#Render[5]..<SNR>301_Render[18]..<SNR>301_DescendantsOf:
line    8:
E735: Can only compare Dictionary with Dictionary
Error detected while processing function <SNR>293_on_exit[6]..<SNR>293_ApplyExtracted[14]..vista#viewer#Display[1]..vista#sidebar#OpenOrUpdate[24]..vista#util#SetBufline[3]..<SNR>295_PrependFpath:
line    8:
E745: Using a List as a Number
Error detected while processing function <SNR>293_on_exit[6]..<SNR>293_ApplyExtracted[14]..vista#viewer#Display[1]..vista#sidebar#OpenOrUpdate[24]..vista#util#SetBufline[3]..<SNR>295_PrependFpath:
line    8:
E15: Invalid expression: [fpath, ''] + a:lines
```

**Screenshot or gif** (if possible)
![image](https://user-images.githubusercontent.com/1149255/56867870-b66bd380-69f3-11e9-85d7-07ad90849f68.png)

**Comment by liuchengxu** [comment]
Thanks for the report. Can you try [this branch](https://github.com/liuchengxu/vista.vim/tree/fix-descendents-of)? Should be a bug from #58.

**Comment by idanarye** [comment]
Works fine with that branch.
